esp/services/WsDeploy/WsDeployService.cpp

```diff
@@ -4665,7 +4665,13 @@ bool CWsDeployFileInfo::handleEspServiceBindings(IEspContext &context, IEspHandl
         if (resource)
           xpath.appendf("[@resource='%s']", resource);
 
-        IPropertyTree* pSubType = pEnvRoot->queryPropTree(xpath.str());
+        IPropertyTree* pSubType = nullptr;
+        Owned<IPropertyTreeIterator> iterSubType = pEnvRoot->getElements(xpath.str());
```

Comment:
I suspect we really need to fix both the symptom (crashing if there is more than one element matching) and the cause (whatever allowed the duplicate entries to be created in the first place). Either fix by itself is insufficient...